components/locid_transform/src/expander.rs

```diff
@@ -521,18 +561,36 @@ impl LocaleExpander {
             return update_langid_minimize(max.language, None, None, langid);
         }
 
-        trial.script = None;
-        trial.region = max.region;
-        self.maximize(&mut trial);
-        if trial == max {
-            return update_langid_minimize(max.language, None, max.region, langid);
-        }
+        if favor_region {
+            trial.script = None;
+            trial.region = max.region;
+            self.maximize(&mut trial);
 
-        trial.script = max.script;
-        trial.region = None;
-        self.maximize(&mut trial);
-        if trial == max {
-            return update_langid_minimize(max.language, max.script, None, langid);
+            if trial == max {
+                return update_langid_minimize(max.language, None, max.region, langid);
+            }
+
+            trial.script = max.script;
+            trial.region = None;
+            self.maximize(&mut trial);
+            if trial == max {
+                return update_langid_minimize(max.language, max.script, None, langid);
+            }
+        } else {
+            trial.script = max.script;
+            trial.region = None;
+            self.maximize(&mut trial);
+            if trial == max {
+                return update_langid_minimize(max.language, max.script, None, langid);
+            }
+
+            trial.script = None;
+            trial.region = max.region;
+            self.maximize(&mut trial);
+
+            if trial == max {
+                return update_langid_minimize(max.language, None, max.region, langid);
+            }
         }
```

Comment:
I've duplicated some of this code instead of extracting it into a closure (like the commits before rebasing) since it seems be add more complexity that it feels to be worth.